pkg/query/storeset.go

```diff
@@ -143,13 +143,13 @@ type storeSetNodeCollector struct {
 	connectionsDesc *prometheus.Desc
 }
 
-func newStoreSetNodeCollector() *storeSetNodeCollector {
+func newStoreSetNodeCollector(configInstance string) *storeSetNodeCollector {
 	return &storeSetNodeCollector{
 		storeNodes: map[component.StoreAPI]map[string]int{},
 		connectionsDesc: prometheus.NewDesc(
 			"thanos_store_nodes_grpc_connections",
 			"Number of gRPC connection to Store APIs. Opened connection means healthy store APIs available for Querier.",
-			[]string{"external_labels", "store_type"}, nil,
+			[]string{"external_labels", "store_type"}, map[string]string{"config_instance": configInstance},
```

Comment:
What is config instance?

Reply:
As we are iterating over the config list, so for more than one list item, it is trying to register a new metrics collector and was throwing the error `panic: duplicate metrics collector registration attempted`, so to make each metrics collector unique, we've added a `name` option in endpoint config.

The other option was to use the config item number to make each metric collector unique. @kakkoyun suggested taking a unique name from the user (for each config) so that in case there is some bad configuration set from the user side, they can debug, and here `name` would be more helpful than a number (e.g., 0, 1) to know where their config is wrong.

The same is the case with the DNS provider, for which it was not giving any specific error but not functioning as expected. For instance, for 2 configs, it was first registering the config[0] endpoints then removing them, saying that "these endpoints are not more active" and registering the config[1] endpoints and then removing these too with the same msg. And now again, adding the config[0] endpoints, and the same process repeats. 
The timely addition and removal is due to the periodical updating of endpoints ["update()"](https://github.com/thanos-io/thanos/pull/4389/files#diff-f9808a0ac67a474a18f6a7ea767593f85c216aeafe322c1ff5d17c81ababff95R476)

Reply:
Maybe to avoid any confusion call it `provider_name` or `config_provider_name`?